File: packages/js/core/src/uri-resolution/core/types/InternalResolverError.ts
import { ResolveUriErrorType } from ".";
import { ResolveUriError } from "./ResolveUriError";

export class InternalResolverError implements ResolveUriError {
  type: ResolveUriErrorType = ResolveUriErrorType.InternalResolver;
  resolverName: string;
  error: Error;

  constructor(resolverName: string, error: Error) {
    this.resolverName = resolverName;
    this.error = error;
  }
}

File: packages/py/core/src/uri_resolution/core/types/internal_resolver_error.py
from . import ResolveUriErrorType
from . import ResolveUriError


class InternalResolverError(ResolveUriError):
    type = ResolveUriErrorType.InternalResolver

    def __init__(self, resolver_name: str, error: Exception):
        self.resolver_name = resolver_name
        self.error = error
